ui/pages/settings/advanced-tab/advanced-tab.component.js

```diff
@@ -117,6 +121,111 @@ export default class AdvancedTab extends PureComponent {
     );
   }
 
+  async getTextFromFile(file) {
+    return new Promise((resolve, reject) => {
+      const reader = new window.FileReader();
+      reader.onload = (e) => {
+        const text = e.target.result;
+        resolve(text);
+      };
+
+      reader.onerror = (e) => {
+        reject(e);
+      };
+
+      reader.readAsText(file);
+    });
+  }
+
+  async handleFileUpload(event) {
+    const file = event.target.files[0];
+    const jsonString = await this.getTextFromFile(file);
+    const result = await this.props.restoreUserData(jsonString);
+    this.setState({
+      showResultMessage: true,
+      restoreSuccessful: result,
+    });
+  }
+
+  renderRestoreUserData() {
+    const { t } = this.context;
+    const { showResultMessage, restoreSuccessful } = this.state;
+
+    return (
+      <div
+        ref={this.settingsRefs[0]}
```

Comment:
```suggestion
        ref={this.settingsRefs[15]}
```
The index added currently is a duplicate value as 0 is used for the state log entry